File: Testowanie i Dokumentowanie Aplikacji/klasa 4/tida testy jednostkowe 3 języki/Kosc.cs
namespace konsolowa;

public class Kosc
{
    public static int Instancje;

    private static readonly Random _rng = new();

    public List<string> NazwyPlikowObrazow { get; set; }
    public int LiczbaOczek { get; set; }
    public int IdentyfikatorPliku { get; set; }
    public bool Dostepna { get; set; }

    public Kosc(int wartosc)
    {
        NazwyPlikowObrazow = new List<string>();
        for (var i = 0; i <= 6; i++)
            NazwyPlikowObrazow.Add($"kosc{i}.png");

        if (!(wartosc >= 1 && wartosc <= 6)) wartosc = 0;

        LiczbaOczek = wartosc;
        IdentyfikatorPliku = wartosc;
        Dostepna = true;
        Instancje++;
    }

    public Kosc()
    {
        NazwyPlikowObrazow = new List<string>();
        for (var i = 0; i <= 6; i++)
            NazwyPlikowObrazow.Add($"kosc{i}.png");

        LiczbaOczek = _rng.Next(1, 7);
        IdentyfikatorPliku = LiczbaOczek;
        Dostepna = true;
        Instancje++;
    }

    public void Rzut()
    {
        if (!Dostepna)
            return;

        LiczbaOczek = _rng.Next(1, 7);
        IdentyfikatorPliku = LiczbaOczek;
    }

    public void Blokuj()
    {
        Dostepna = false;
    }

    public string PobierzRzutTekst()
    {
        if (!(LiczbaOczek >= 1 && LiczbaOczek <= 6)) return "nie ustawiono";
        string[] liczbySłownie = { "jeden", "dwa", "trzy", "cztery", "pięć", "sześć" };
        return liczbySłownie[LiczbaOczek - 1];
    }

    public string PobierzNazwePliku()
    {
        if (IdentyfikatorPliku >= 0 && IdentyfikatorPliku < NazwyPlikowObrazow.Count)
            return NazwyPlikowObrazow[IdentyfikatorPliku];
        return string.Empty;
    }
}

File: Testowanie i Dokumentowanie Aplikacji/klasa 4/tida testy jednostkowe 3 języki/Kosc.py
import random


class Kosc:
    Instancje = 0
    _rng = random.Random()

    def __init__(self, wartosc=None):
        self.NazwyPlikowObrazow = [f"kosc{i}.png" for i in range(0, 7)]

        if wartosc is None:
            self.LiczbaOczek = Kosc._rng.randint(1, 6)
        else:
            if 1 <= wartosc <= 6:
                self.LiczbaOczek = wartosc
            else:
                self.LiczbaOczek = 0

        self.IdentyfikatorPliku = self.LiczbaOczek
        self.Dostepna = True
        Kosc.Instancje += 1

    def Rzut(self):
        if not self.Dostepna:
            return
        self.LiczbaOczek = Kosc._rng.randint(1, 6)
        self.IdentyfikatorPliku = self.LiczbaOczek

    def Blokuj(self):
        self.Dostepna = False

    def PobierzRzutTekst(self):
        if not (1 <= self.LiczbaOczek <= 6):
            return "nie ustawiono"
        liczby_słownie = ["jeden", "dwa", "trzy", "cztery", "pięć", "sześć"]
        return liczby_słownie[self.LiczbaOczek - 1]

    def PobierzNazwePliku(self):
        if 0 <= self.IdentyfikatorPliku < len(self.NazwyPlikowObrazow):
            return self.NazwyPlikowObrazow[self.IdentyfikatorPliku]
        return ""
